rfcs/2021-08-29-8381-vrl-iteration-support.md

```diff
@@ -90,28 +102,380 @@ One gap in the language right now is the possibility to _dynamically remap
+
+   ```coffee
+   array =  [{ "a": 2}, {"a": 3}]
+   any_two = false
+   for_each(array) -> |_index, value| { if value == 2 { any_two = true } }
+   ```
+
+   **NOTE** This is a good use-case for `any` and `all` iteration functions:
+
+   ```coffee
+   any_two = any(array) -> |_index, value| { value == 2 }
+   any_two = all(array) -> |_index, value| { value != 2 }
+   ```
+
+6. [call `parse_timestamp` on array of Cloudtrail records](https://discord.com/channels/742820443487993987/746070604192415834/919984998482870302)
+
+   ```coffee
+   . = [{ ... }, { ... }]
+   . = map_values(.) -> |value| {
+     value.timestamp = parse_timestamp(value.eventTime, "%Y-%m-%dT%H:%M:%SZ") ?? now()
+     value
+   }
+   ```
+
+7. [”unzip” object into separate key/value arrays](https://discord.com/channels/742820443487993987/746070591097798688/915227297697628190)
+
+   ```coffee
+   keys = []
+   values = []
+
+   for_each(.) -> |key, value| {
+     keys = push(keys, key)
+     values = push(values, value)
+   }
+   ```
+
+8. [add fields to objects in array](https://discord.com/channels/742820443487993987/764187584452493323/914082502149283872)
+
+   ```coffee
+   . = { "foo": "bar", "items": [{}, {}] }
+   .items = map_values(.items) -> |value| { value.foo = .foo; value }
+   ```
+
+9. ["zip" an array of objects with fields `key` and `value` into one object](https://discord.com/channels/742820443487993987/764187584452493323/905803048851505174)
+
+   ```coffee
+   data = [{ "key": "name", "value": "value" }, { "key": "key", "value": "otherValue" }]
+   for_each(data) -> |_index, value| {
+     . = set(., [value.key], value.value)
+   }
+   ```
+
+10. [trim a character from all keys](https://discord.com/channels/742820443487993987/746070591097798688/905799877492101121)
+
+   ```coffee
+   . = map_keys(., recursive: true) -> |key| { trim_start(key, "_") }
+   ```
+
+11. [add prefix to all keys](https://discord.com/channels/742820443487993987/764187584452493323/883274684576182302)
+
+   ```coffee
+   . = map_keys(., recursive: true) -> |key| { "my_" + key }
+   ```
+
+
+12. [parse message using list of Grok patterns until one matches](https://discord.com/channels/742820443487993987/764187584452493323/870353108692271104)
+
+   ```coffee
+   patterns = []
+   matched = false
+
+   for_each(patterns) -> |_index, pattern| {
+     if !matched && (parsed, err = parse_grok(.message, pattern); err == null) {
+       matched = true
+       . |= parsed
+     }
+   }
+   ```
+
+13. [find match against list of regular expressions](https://discord.com/channels/742820443487993987/764187584452493323/864496206947942400)
+
+   ```coffee
+   matched = false
+   for_each(patterns) -> |pattern| {
+     if !matched && match(.message, pattern) {
+       matched = true
+     }
+   }
+   ```
+
+   **NOTE** this would be less verbose (and slightly more performant) using
+   a future `any` function:
+
+   ```coffee
+   matched = any(patterns) -> |pattern| { match(.message, pattern) }
+   ```
```

Comment:
It would be nice to extract all of these future "shortcut" function into the "Future Work" section. This way we can come back to it as follow up work.